core/capabilities/ccip/ccipaptos/extradatadecoder.go

```diff
@@ -0,0 +1,201 @@
+package ccipaptos
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+)
+
+// ExtraDataDecoder is a concrete implementation of ccipcommon.ExtraDataDecoder
+// Compatible with ccip::fee_quoter version 1.6.0
+type ExtraDataDecoder struct{}
+
+var _ ccipcommon.ExtraDataDecoder = (*ExtraDataDecoder)(nil)
+
+const (
+	aptosDestExecDataKey = "destGasAmount"
+)
+
+var (
+	// bytes4 public constant EVM_EXTRA_ARGS_V1_TAG = 0x97a657c9;
+	evmExtraArgsV1Tag = hexutil.MustDecode("0x97a657c9")
+
+	// bytes4 public constant GENERIC_EXTRA_ARGS_V2_TAG = 0x181dcf10;
+	genericExtraArgsV2Tag = hexutil.MustDecode("0x181dcf10")
+
+	// bytes4 public constant SVM_EXTRA_EXTRA_ARGS_V1_TAG = 0x1f3b3aba
+	svmExtraArgsV1Tag = hexutil.MustDecode("0x1f3b3aba")
```

Comment:
Looks like these tags are duplicated from `ccipevm`, bonus if we can consolidate in a common location, perhaps `core/capabilities/ccip/common`